### ISSUE TITLE
Question about fixed std=0.02 initialization of `w1` in `moe.py`

### ISSUE THREAD

**Issue by trestad** [pytorch/torchtitan#1257]
Hi torchtitan team,

Thanks for the great work on this project! I had a question regarding a detail in the code at moe.py#L92

https://github.com/pytorch/torchtitan/blob/768cde131105bde624160029d808e94649faf0f4/torchtitan/experiments/llama4/model/moe.py#L92

I noticed that `w1` is initialized with a fixed standard deviation of 0.02, whereas `w2` and `w3` are initialized using a configurable `init_std` parameter. I’m wondering if this discrepancy is intentional, and if so, what the reasoning is behind using a hardcoded value for `w1`.

Would greatly appreciate any insights you could share!

Thanks again!

**Comment by tianyu-l** [comment]
I copied this from Llama 3 FFN init code at https://github.com/pytorch/torchtitan/blob/main/torchtitan/models/llama3/model.py#L305

@lessw2020 do you have more context on the choice of `std` for `w1`, `w2`, `w3`?

**Comment by sippycoder** [comment]
If you go for spectral conditioning, for the usual 2048 dim, the standard deviation comes close to 0.02. 

https://arxiv.org/pdf/2310.17813

However, I also don't understand how initializing just `w1` with 0.02 and rest with depth based initialization would make sense for any general model scale. I'm also waiting for the reply for the above question. Is there a variance-based mathematical explanation?

**Comment by lessw2020** [comment]
Hi @sippycoder and @trestad,
Quick background on the init numbers - this was based on work we were doing two+ years ago for parallel transformer blocks and also the Olmo paper.   We had tested at the time with with and without the layer/depth based init and results were similar but seemed to slightly favor depth based so that's what we went with. 

We haven't looked at updating since then as it has worked well, but happy to take any PR's if you have potential improvements.

**Comment by lessw2020** [comment]
The spectral conditioning paper linked above looks promising but odd that they didn't test on anything larger than a cifar-10 size model? 
Looks like implementation would be pretty straightforward though we also need to modify the applied learning rates per layer which might add a small amount of overhead. 
@sippycoder  - are you interested in doing a PR to test this out? 

<img width="909" height="864" alt="Image" src="https://github.com/user-attachments/assets/4155f792-abdc-486d-bfdf-22def6d97fe5" />
